server/grpc_service.go

```diff
@@ -1680,7 +1680,10 @@ func (s *GrpcServer) BatchScanRegions(ctx context.Context, request *pdpb.BatchSc
 		}
 		keyRanges.Append(reqRange.StartKey, reqRange.EndKey)
 	}
-	res := rc.BatchScanRegions(keyRanges, int(limit))
+	res, err := rc.BatchScanRegions(keyRanges, int(limit), request.OutputMustContainAllKeyRange)
+	if err != nil {
```

Comment:
I'm not sure this error check relies too much on the behaviour of `rc.BatchScanRegions`. What if it will return another type of error (not ErrorType_REGIONS_NOT_CONTAIN_ALL_KEY_RANGE) in future?

Reply:
In that case, we only have to do some more complicated error handling :(

Reply:
In future the developer may forget to change this line. So I sightly recommend at present we add more strict check before converting errors.